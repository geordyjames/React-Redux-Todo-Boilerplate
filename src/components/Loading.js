import React from 'react';

const Loading = props => {
	if (props.error) {
		return (
			<div className="uk-flex uk-flex-center uk-flex-middle half-height">
				Error!
				<button
					className="uk-button uk-button-danger uk-text-bold uk-margin-left"
					onClick={props.retry}
				>
					Retry
				</button>
			</div>
		);
	} else if (props.timedOut) {
		return (
			<div className="uk-flex uk-flex-center uk-flex-middle half-height">
				Taking a long time...
				<button
					className="uk-button uk-button-danger uk-text-bold uk-margin-left"
					onClick={props.retry}
				>
					Retry
				</button>
			</div>
		);
	} else if (props.pastDelay) {
		return (
			<div className="uk-flex uk-flex-center uk-flex-middle half-height">
				<div data-uk-spinner="ratio: 3" />
			</div>
		);
	} else {
		return null;
	}
};

export default Loading;
