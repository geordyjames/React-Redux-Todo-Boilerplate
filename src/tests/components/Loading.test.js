import React from 'react';
import { shallow } from 'enzyme';
import Loading from '../../components/Loading';

test('should render Loading with empty props correctly', () => {
  const wrapper = shallow(<Loading />);
  expect(wrapper).toMatchSnapshot();
});

test('should render Loading with error prop correctly', () => {
  const wrapper = shallow(<Loading error={true} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render Loading with timedOut prop correctly', () => {
  const wrapper = shallow(<Loading timedOut={true} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render Loading with pastDelay prop correctly', () => {
  const wrapper = shallow(<Loading pastDelay={true} />);
  expect(wrapper).toMatchSnapshot();
});