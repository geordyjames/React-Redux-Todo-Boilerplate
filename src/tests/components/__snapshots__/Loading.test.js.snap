// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Loading with empty props correctly 1`] = `""`;

exports[`should render Loading with error prop correctly 1`] = `
<div
  className="uk-flex uk-flex-center uk-flex-middle half-height"
>
  Error!
  <button
    className="uk-button uk-button-danger uk-text-bold uk-margin-left"
  >
    Retry
  </button>
</div>
`;

exports[`should render Loading with pastDelay prop correctly 1`] = `
<div
  className="uk-flex uk-flex-center uk-flex-middle half-height"
>
  <div
    data-uk-spinner="ratio: 3"
  />
</div>
`;

exports[`should render Loading with timedOut prop correctly 1`] = `
<div
  className="uk-flex uk-flex-center uk-flex-middle half-height"
>
  Taking a long time...
  <button
    className="uk-button uk-button-danger uk-text-bold uk-margin-left"
  >
    Retry
  </button>
</div>
`;
